runtime/src/mimalloc/c/include/mimalloc-internal.h

```diff
@@ -221,27 +234,120 @@ static inline size_t _mi_wsize_from_size(size_t size) {
   return (size + sizeof(uintptr_t) - 1) / sizeof(uintptr_t);
 }
 
+// Does malloc satisfy the alignment constraints already?
+static inline bool mi_malloc_satisfies_alignment(size_t alignment, size_t size) {
+  return (alignment == sizeof(void*) || (alignment == MI_MAX_ALIGN_SIZE && size > (MI_MAX_ALIGN_SIZE/2)));
+}
 
-/* -----------------------------------------------------------
-  The thread local default heap
------------------------------------------------------------ */
+// Overflow detecting multiply
```

Comment:
We used to have some custom handling for `mi_mul_overflow` https://github.com/JetBrains/kotlin-native/blob/19317fa7269f2f17624e043368469f0fa6e1c568/runtime/src/mimalloc/c/include/mimalloc-internal.h#L169 is it not needed anymore?

Reply:
It's needed, I now made another hack, better to use this one, you're right. Fixed.